base/libc.jl

```diff
@@ -137,7 +137,7 @@ systemsleep
 
 struct TimeVal
```

Comment:
Do we consider this breaking? If so, we may want a getproperty overload that provides access to the old field name.

Reply:
I don't think so - it's neither exported, nor does it show up on docs.julialang.org (as far as I can see).

Reply:
It's not exported from Base, but `Libc` is and people are using these as `Libc.TimeVal`, e.g.:
https://github.com/ancapdev/LightBSON.jl/blob/729b4308793b26dd4fa6635c224b8165299ff6a4/src/object_id.jl#L69
https://github.com/ancapdev/UnixTimes.jl/blob/master/src/UnixTimes.jl#L104
https://github.com/timholy/Revise.jl/blob/master/src/utils.jl#L111-L115

Reply:
Right, I did change the type from `timeval` to `timespec` - I should leave that struct alone and add `TimeSpec` to `Libc` instead, with a conversion from `TimeSpec` (and an deprecation notice for `TimeVal`?)

How did you come up with that list? I think all of these should have used `time()` in the first place, rather than manually calling into `Libc` on their own :thinking: I may make PRs fixing that, if the list isn't too long..

Reply:
Trying that deprecation, I don't think I can easily add that do `deprecate.jl` using `@deprecate` since that defines a new method, which I don't necessarily want if I want to "hook into" the new `TimeSpec` and just convert it - how should I go about that?

Reply:
Well, I can't deprecate the constructor, since `depwarn` only takes a symbol, not a signature...

Reply:
Ok, UnixTimes and LightBSON can be patched/worked around once this PR hits, Revise has a PR open now and I'll open an issue for InterProcessCommunication.jl once this hits. The struct has to remain for now, but at least one constructor is marked deprecated. I'll add the deprecation notice of that to `NEWS.md` as well.